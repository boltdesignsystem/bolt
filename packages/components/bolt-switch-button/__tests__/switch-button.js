import { render, stopServer } from '../../../testing/testing-helpers';
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    label: 'Toggle me',
    attributes: {
      class: 'js-target-the-switch-button-container',
    },
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Switch Button', () => {
  test(`default`, async () => {
    const results = await render(
      '@bolt-components-switch-button/switch-button.twig',
      {
        ...fixtures.defaultData,
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt Switch Button Prop', () => {
  test(`button_attributes`, async () => {
    const results = await render(
      '@bolt-components-switch-button/switch-button.twig',
      {
        ...fixtures.defaultData,
        button_attributes: {
          class: 'switch-button-class-example',
        },
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`on`, async () => {
    const results = await render(
      '@bolt-components-switch-button/switch-button.twig',
      {
        ...fixtures.defaultData,
        on: false,
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`custom id`, async () => {
    const results = await render(
      '@bolt-components-switch-button/switch-button.twig',
      {
        ...fixtures.defaultData,
        button_attributes: {
          id: 'a-uuid-test-12345',
        },
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});
