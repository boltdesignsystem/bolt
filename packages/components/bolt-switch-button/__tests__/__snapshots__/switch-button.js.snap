// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Switch Button Prop button_attributes 1`] = `
<label class="js-target-the-switch-button-container c-bolt-switch-button"
       for="bolt-switch-button-test-target"
       uuid="12345"
>
  <div class="c-bolt-switch-button__label">
    Toggle me
  </div>
  <button class="switch-button-class-example c-bolt-switch-button__button"
          type="button"
          id="bolt-switch-button-test-target"
          role="switch"
          aria-checked="true"
  >
    <span class="c-bolt-switch-button__button-text c-bolt-switch-button__button-text--checked">
      on
    </span>
    <span class="c-bolt-switch-button__button-text c-bolt-switch-button__button-text--unchecked ">
      off
    </span>
  </button>
</label>
`;

exports[`Bolt Switch Button default 1`] = `
<label class="js-target-the-switch-button-container c-bolt-switch-button"
       for="bolt-switch-button-test-target"
       uuid="12345"
>
  <div class="c-bolt-switch-button__label">
    Toggle me
  </div>
  <button class="c-bolt-switch-button__button"
          type="button"
          id="bolt-switch-button-test-target"
          role="switch"
          aria-checked="true"
  >
    <span class="c-bolt-switch-button__button-text c-bolt-switch-button__button-text--checked">
      on
    </span>
    <span class="c-bolt-switch-button__button-text c-bolt-switch-button__button-text--unchecked ">
      off
    </span>
  </button>
</label>
`;
